core/arch/arm/kernel/kern.ld.S

```diff
@@ -356,3 +378,8 @@ SECTIONS
 	}
 
 }
+
+#if defined(CFG_TEE_COHERENT_START)
+ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
+ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
```

Comment:
Adding several ASSERT() inside the macro `DECLARE_COHERENT_RAM_SECTION` combine with the 80 chars max line size make the macro not very easy to read. I will move all assertion at the end of the file. Nevertheless, fell free to comment back...